tools/ci_build/build.py

```diff
@@ -373,6 +378,25 @@ def generate_build_tree(cmake_path, source_dir, build_dir, cuda_home, cudnn_home
         cmake_args += ["-DCMAKE_TOOLCHAIN_FILE=" + args.android_ndk_path + "/build/cmake/android.toolchain.cmake",
                 "-DANDROID_PLATFORM=android-" + str(args.android_api),
                 "-DANDROID_ABI=" + str(args.android_abi)]
+    
+    if args.ios:
+        needed_args = [len(args.ios_sysroot) == 0, not args.arm64 and not args.arm, len(args.ios_toolchain_dir) == 0]
+        arg_names = ["--ios_sysroot <path to sysroot>", "--arm or --arm64", "--ios_toolchain_dir <path to toolchain>"]
+        if any(needed_args):
+            raise BuildError("iOS build canceled due to missing arguments: " + (val + "," for val, cond in zip(arg_names, needed_args) if cond))
+        compilers = sorted(glob.glob(args.ios_toolchain_dir + "/bin/*-clang*"))
+        os.environ["PATH"] = args.ios_toolchain_dir + "/bin:" + os.environ.get("PATH", "")
+        os.environ["LD_LIBRARY_PATH"] = args.ios_toolchain_dir + "/lib:" + os.environ.get("LD_LIBRARY_PATH", "")
+        #print(compilers)
+        #exit(0)
+        if len(compilers) != 2:
+            raise BuildError("error identifiying compilers in toolchaindir")
```

Comment:
identifiying->identifying, toolchaindir->toolchain_dir?

Reply:
corrected typos
I'm not sure if the way I check for the compilers, does work generally for different toolchains.
I assumed that two instances of files matching "/*clang*" do exist in the bin folder of the ios_toochain_dir